src/colmap/estimators/cost_functions.h

```diff
@@ -47,27 +47,83 @@ template <typename T>
 using EigenQuaternionMap = Eigen::Map<const Eigen::Quaternion<T>>;
 using EigenMatrix6d = Eigen::Matrix<double, 6, 6>;
 
+enum CovarianceType {
```

Comment:
IDENTITY, ISOTROPIC, ANISOTROPIC ?

Reply:
Let's make it an enum class to gain better type safety?

Reply:
Sure. I can make it a class. DIAGONAL supports not only isotropic covariance. The diagonal values can differ which is anisotropic by definition

Reply:
I already edited the comment. ANISOTROPIC might still be the better term than GENERAL?

Reply:
ANISOTROPIC does not feel accurate here. How about DENSE?

Reply:
I am fine with "dense" but imo "anisotropic" is also correct from a statistical standpoint, no?